src/goMain.ts

```diff
@@ -118,15 +122,23 @@ export function activate(ctx: vscode.ExtensionContext): void {
 				}
 			);
 
-			ctx.subscriptions.push(c.start());
-
 			c.onReady().then(() => {
+				const capabilities = c.initializeResult && c.initializeResult.capabilities;
+				if (capabilities && !capabilities.completionProvider) {
+					ctx.subscriptions.push(vscode.languages.registerCompletionItemProvider(GO_MODE, new GoCompletionItemProvider(), '.', '\"'));
+					vscode.window.showInformationMessage('Your installed version of `go-langserver` is out of date and does not support code completion. Falling back to default behavior.');
```

Comment:
@ramya-rao-a Is this an ok phrasing or would you also prompt the user to upgrade? It might also make sense to show a single generic message in case _any_ fallback condition inside this function body was triggered.

Reply:
@m90 I'll look into the update option later. I've pushed a commit to use generic message